include/bluetooth/buf.h

```diff
@@ -69,12 +69,13 @@ struct net_buf *bt_buf_get_cmd_complete(s32_t timeout);
  *  This will set the buffer type so bt_buf_set_type() does not need to
  *  be explicitly called before bt_recv_prio() or bt_recv().
  *
- *  @param evt     HCI event code
- *  @param timeout Timeout in milliseconds, or one of the special values
- *                 K_NO_WAIT and K_FOREVER.
+ *  @param evt          HCI event code
+ *  @param discardable  Whether the driver considers the event discardable.
+ *  @param timeout      Timeout in milliseconds, or one of the special values
+ *                      K_NO_WAIT and K_FOREVER.
  *  @return A new buffer.
  */
-struct net_buf *bt_buf_get_evt(u8_t evt, s32_t timeout);
+struct net_buf *bt_buf_get_evt(u8_t evt, bool discardable, s32_t timeout);
```

Comment:
I assume the K_NO_WAIT would discard anyway so I not sure why we need another variable besides that is completely ignore when CONFIG_BT_DISCARDABLE_POOL is not set.

Reply:
@Vudentz those are not quite the same. A HCI driver might be calling this from an ISR for a non-discardable event, in which case K_NO_WAIT is the right timeout to pass. E.g. the h4.c driver does this, and then if that fails it defers the further allocation attempts to the RX thread.

Reply:
Fair enough, I though would use the new pool as well but it seems they are exclusive to each type.

Reply:
Yeah, I'm all for eliminating redundancy in parameters, but in this case they're a bit orthogonal and we can't make too many assumptions about getting a K_NO_WAIT.